ipfx/nwb_writer.py

```diff
@@ -0,0 +1,71 @@
+import pynwb
+from allensdk.core.nwb_data_set import NwbDataSet
+import ipfx.nwb_reader as nwb_reader
+
+
+class NwbWriter(object):
+
+    def __init__(self, nwb_file_name):
+        self.nwb_file_name = nwb_file_name
+
+    def add_spike_times(self, sweep_num, spike_times):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+
+class Nwb1Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+        self.nwbfile = NwbDataSet(self.nwb_file_name)
+
+    def add_spike_times(self, sweep_num, spike_times):
+        self.nwbfile.set_spike_times(sweep_num, spike_times)
+
+    def save(self):
+        pass
+
+
+class Nwb2Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+
+        io = pynwb.NWBHDF5IO(self.nwb_file_name, 'a')
+        self.nwbfile = io.read()
```

Comment:
I think this will fail if the nwbfile does not already exist. OK for your embed_spike_times use case, but a bit odd for a class called `Nwb2Writer`

Reply:
I assume pynwb follows h5py's modes with "a": Read/write if exists, create otherwise (default)

Reply:
I think the `io.read` call will error out if the file does not already exist.

Reply:
Is this ok to be append-only?

Reply:
You are right again, `io.read` errors out if file does not exist.
I turned the class into `NwbAppender` and raise `FineNotFoundError` exception if file does not exist.